modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/coverage/io/netcdf/crs/NetCDFCoordinateReferenceSystemType.java

```diff
@@ -161,31 +164,16 @@ public static NetCDFCoordinateReferenceSystemType parseCRS(CoordinateReferenceSy
             crsType = WGS84;
         } else if (crs instanceof ProjectedCRS) {
             ProjectedCRS projectedCRS = (ProjectedCRS) crs;
-            Projection projection = projectedCRS.getConversionFromBase();
-            MathTransform transform = projection.getMathTransform();
-            if (transform instanceof TransverseMercator) {
-                crsType = TRANSVERSE_MERCATOR;
-            } else if (transform instanceof LambertConformal1SP) {
-                crsType = LAMBERT_CONFORMAL_CONIC_1SP;
-            } else if (transform instanceof LambertConformal2SP) {
-                crsType = LAMBERT_CONFORMAL_CONIC_2SP;
-            } else if (transform instanceof LambertAzimuthalEqualArea) {
-                crsType = LAMBERT_AZIMUTHAL_EQUAL_AREA;
-            } else if (transform instanceof Orthographic) {
-                crsType = ORTHOGRAPHIC;
-            } else if (transform instanceof PolarStereographic) {
-                crsType = POLAR_STEREOGRAPHIC;
-            } else if (transform instanceof Stereographic) {
-                crsType = STEREOGRAPHIC;
-            } else if (transform instanceof Mercator1SP) {
-                crsType = MERCATOR_1SP;
-            } else if (transform instanceof Mercator2SP) {
-                crsType = MERCATOR_2SP;
-            } else if (transform instanceof AlbersEqualArea) {
-                crsType = ALBERS_EQUAL_AREA;
-            } else if (transform instanceof RotatedPole) {
-                crsType = ROTATED_POLE;
+            // NetCDF supports a reduced number of CRS Type (WGS84 + some projection).
+            // Some ProjectedCRS may have unit of measure resulting in concatenated transforms.
+            if (hasConcatenatedTransform(projectedCRS)) {
+                Projection projection = projectedCRS.getConversionFromBase();
+                MathTransform mt = projection.getMathTransform();
+                crsType = getProjectionType(((ConcatenatedTransform) mt).transform1);
```

Comment:
How do you know, in general, that the concatenated transform you need to look at is the first? In theory there is a chain of N (multple concatenated transforms) and the projection steps. But I agree that's the common case, and maybe the only case currently implemented in the code base, but in general there will be many... maybe add a check that there are only two, and if one of them is a concatenated itself, throw a meaningful exception? (something like "we dont' support this case").

Reply:
Right. Will do. thanks